superset/connectors/druid/models.py

```diff
@@ -1237,15 +1237,21 @@ def query(self, query_obj):
 
         # Reordering columns
         cols = []
+        groupby = query_obj.get('groupby') or []
+        columns = query_obj.get('columns') or []
+        metrics = query_obj.get('metrics') or []
         if DTTM_ALIAS in df.columns:
             cols += [DTTM_ALIAS]
-        cols += query_obj.get('groupby') or []
-        cols += query_obj.get('columns') or []
-        cols += query_obj.get('metrics') or []
+        cols += groupby
+        cols += columns
+        cols += metrics
 
         cols = [col for col in cols if col in df.columns]
         df = df[cols]
 
+        for col in groupby + columns:
```

Comment:
would it be possible for some of these columns to not be in df.columns given the list comprehension at line 1249?